pkg/controller/workload/workload_processor.go

```diff
@@ -188,15 +188,28 @@ func (p *Processor) removeWorkloadFromBpfMap(uid string) error {
 	return nil
 }
 
-func (p *Processor) deleteFrontendData(id uint32) error {
+func (p *Processor) deleteServiceFrontendData(service *workloadapi.Service, id uint32) error {
 	var (
 		err error
 		fk  = bpf.FrontendKey{}
 	)
+
+	// If old service exist, use its address
+	if service != nil {
+		for _, networkAddress := range service.GetAddresses() {
+			nets.CopyIpByteFromSlice(&fk.Ip, networkAddress.Address)
+			if err = p.bpf.FrontendDelete(&fk); err != nil {
+				log.Errorf("delete service %s frontend key %v, err: %v", service.ResourceName(), fk, err)
+			}
+		}
+		return nil
+	}
+
+	// Otherwise fall back to iterating over the map, this can only occur on restart
 	if fks := p.bpf.FrontendIterFindKey(id); len(fks) != 0 {
```

Comment:
What kind of scenarios will go to this branch?

Reply:
ok, I got it, `p.compareWorkloadAndServiceWithHashName` will trigger this branch.

Reply:
Right

Reply:
But the name of this function is not good...